doc/source/extended_documentation/regrid/bilinear_land_sea.rst

```diff
@@ -0,0 +1,132 @@
+################################################
+Bilinear regridding with land-sea mask awareness
+################################################
+
+This subpackage provides a fast and unified implementation of bilinear regridding with land-sea mask awareness. It is assumed that the source grid is evenly-spaced in a rectilinear latitude/longitude system. The target grid could be rectilinear or curvilinear. 
+
+********************************
+Calculation steps of Regridding 
+********************************
+ 
+* Given a target point, determine its four source points according to regular grid point ordering.
+
+* Identify surface type of source/target points from land sea mask input files.
+
+* Identify if source points have the same surface type as the target point. 
+  The matched source points are called "True", and unmatched source points are called "False". 
+
+* Calculate weights of four source points with normal bilinear formulation.
+
+* Adjust source points & weights if unmatched source points exist. 
+
+* Apply weights at values of source points for interpolated value in a unified way
+  so that vectorisation of the Numpy computation can be fully achieved.
+  
+*************************************
+The weight calculation and adjustment
+*************************************
+
+::
+                        
+                4   .----------. 3     
```

Comment:
I've been trying to work out if this matches with the numberic number assigned in the code. I can't quite convince myself either way, I *think* :

Either the label of  lat / lon need flipping such that lon is along the bottom of the cells?

_or_

The numeric ordering might be different from that assigned  in 
 https://github.com/metoppv/improver/blob/e7240bd57c4770033328dcdbf0eb13118edfff30/improver/regrid/bilinear.py#L105

What was the intention here Julian? From the text it thought it was showing the spatial layout of source points, which should corespond to the labelling of the indexes.

Reply:
Sorry for confusing you. This line means:  1D ordering  (lat1, lon1)(lat1,lon2),(lat1,lon3),(lat1,lon4),(lat1,lon5) ....(lat1,lon_last),(lat2,lon1) ....  Here all indexes are referred to the mesh(grid)-level, and different from element-level indexing (1,2,3,4 in figure). At the elemental level, the weighting formulation is the same as long as side 12 is lat (bottom-left point is always 1), i.e. you can assume side12 is vertical or horizontal.  Hope it helps.

Reply:
No need to appologise Julian, just trying to make sure I can understand what the diagram is intending to show. Mainly looking for consistency between the code and the documentation.

The following code has comments that describe the spatial layout of points, it describes point 1 as the lower left, point 2 and the upper left etc. 

https://github.com/metoppv/improver/blob/e7240bd57c4770033328dcdbf0eb13118edfff30/improver/regrid/bilinear.py#L238

It would help massively if the description in the code and the diagrams in the documentation were consistent, allowing a new reader to cross reference the diagram with the code. I was incorrect in how the diagram differed from the description in the code on my first look. Flipping the diagram such that it matches this description would be my recommendation. Such as: 

```
2   .----------. 3
    |          |       target point x(lat,lon)
lat |    x     |       four source points 1,2,3 and 4
    |          |
1   .----------. 4
        lon

````

Reply:
Thanks, I will use this indexing order.   I used it before (as I used in jupyter notebook).  
The only disadvantage is  that (lat,lon) is not consistent with traditional x/y coordinate , which made me change when writing the document.